lms/djangoapps/certificates/apis/v0/views.py

```diff
@@ -120,3 +124,132 @@ def get(self, request, username, course_id):
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        OAuth2AuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+
+    permission_classes = (
+        C(IsAuthenticated) & (
+            C(permissions.NotJwtRestrictedApplication) & (
+                C(permissions.IsStaff) | permissions.IsUserInUrl
+            ) | (
+                C(permissions.JwtRestrictedApplication) &
+                permissions.JwtHasScope &
+                permissions.JwtHasUserFilterForRequestedUser
+            )
+        ),
+    )
+
+    required_scopes = ['certificates:read']
+
+    def get(self, request, username):
+        """
+        Gets the list of viewable course certificates for a specific user.
+
+        Args:
+            request (Request): Django request object.
+            username (string): URI element specifying the user's username.
+
+        Return:
+            A JSON serialized representation of the list of certificates.
+        """
+        user_certs = []
+        for user_cert in self._get_certificates_for_user(username):
+            user_certs.append({
+                'username': user_cert.get('username'),
+                'course_id': unicode(user_cert.get('course_key')),
+                'course_display_name': user_cert.get('course_display_name'),
+                'course_organization': user_cert.get('course_organization'),
+                'certificate_type': user_cert.get('type'),
+                'created_date': user_cert.get('created'),
+                'status': user_cert.get('status'),
+                'is_passing': user_cert.get('is_passing'),
+                'download_url': user_cert.get('download_url'),
+                'grade': user_cert.get('grade'),
+            })
+
+        return Response(user_certs)
+
+    def _get_certificates_for_user(self, username):
+        """
+        Returns a user's viewable certificates sorted by course name.
+        """
+        course_certificates = get_certificates_for_user(username)
+        passing_certificates = {}
+        for course_certificate in course_certificates:
+            if course_certificate.get('is_passing', False):
+                course_key = course_certificate['course_key']
+                passing_certificates[course_key] = course_certificate
+
+        viewable_certificates = []
+        for course_key, course_overview in CourseOverview.get_from_ids_if_exists(passing_certificates.keys()).items():
+            if certificates_viewable_for_course(course_overview):
+                course_certificate = passing_certificates[course_key]
+                course_certificate['course_display_name'] = course_overview.display_name_with_default
+                course_certificate['course_organization'] = course_overview.display_org_with_default
+                viewable_certificates.append(course_certificate)
+
+        viewable_certificates.sort(key=lambda certificate: certificate['course_display_name'])
```

Comment:
I wonder if the default sort is better as descending by `created_date` - so that the more recent certificates appear first.

Reply:
Yes. I agree. This was replicating how we currently display certificates on the existing profile page. I will update the sort order here.

Reply:
@marcotuts FYI - on a choice for default sort order of course certificates.  We're considering created_date over display_name.